src/plugins/discover/public/application/apps/main/services/use_saved_search.ts

```diff
@@ -0,0 +1,328 @@
+              }
+              message.chartData = buildPointSeriesData(tabifiedData, dimensions);
+            }
+          }
+          refs.current.fieldCounts = message.fieldCounts!;
+          refs.current.fetchStatus = message.state;
+          savedSearch$.next(message);
+        },
+        (error) => {
+          if (error instanceof Error && error.name === 'AbortError') return;
+          data.search.showError(error);
+          refs.current.fetchStatus = fetchStatuses.ERROR;
+          savedSearch$.next({
+            state: fetchStatuses.ERROR,
+            inspectorAdapters,
+            fetchError: error,
+          });
+        },
+        () => {
+          refs.current.autoRefreshDoneCb?.();
+          refs.current.autoRefreshDoneCb = undefined;
+        }
+      );
+    },
+    [
+      timefilter,
+      services,
+      stateContainer.appStateContainer,
+      searchSource,
+      indexPattern,
+      useNewFieldsApi,
+      state.hideChart,
+      state.interval,
+      data,
+      searchSessionManager,
+      savedSearch$,
+    ]
+  );
+
+  /**
+   * This part takes care of triggering the data fetching by creating and subscribing
+   * to an observable of various possible changes in state
+   */
+  useEffect(() => {
+    const fetch$ = merge(
+      refetch$,
+      filterManager.getFetches$(),
+      timefilter.getFetch$(),
+      timefilter.getAutoRefreshFetch$().pipe(
+        tap((done) => {
+          refs.current.autoRefreshDoneCb = done;
+        }),
+        filter(() => refs.current.fetchStatus !== fetchStatuses.LOADING)
+      ),
+      data.query.queryString.getUpdates$(),
+      searchSessionManager.newSearchSessionIdFromURL$
+    ).pipe(debounceTime(100));
+
+    const subscription = fetch$.subscribe({
+      next: (val) => {
+        fetchAll(val === 'reset');
+      },
+    });
+
+    return () => {
+      subscription.unsubscribe();
+    };
+  }, [
+    data.query.queryString,
+    filterManager,
+    refetch$,
+    searchSessionManager.newSearchSessionIdFromURL$,
+    timefilter,
+    fetchAll,
+  ]);
+
+  /**
+   * Track state changes that should trigger a fetch
+   */
+  useEffect(() => {
+    let triggerFetch = false;
+    let resetFieldCounts = false;
+    if (state.hideChart !== refs.current.appState.hideChart && !state.hideChart) {
+      // chart was hidden, now it should be displayed, so data is needed
+      triggerFetch = true;
+    } else if (state.interval !== refs.current.appState.interval) {
+      // inverval of chart was changed
+      triggerFetch = true;
+    } else if (!isEqual(state.sort, refs.current.appState.sort)) {
+      // sorting of document table was changed
+      triggerFetch = true;
+    }
+    if (!isEqual(state.index, refs.current.appState.index)) {
+      // different index was selected
+      triggerFetch = true;
+      resetFieldCounts = true;
+    }
+    refs.current.appState = state;
+    if (triggerFetch) {
+      refetch$.next(resetFieldCounts ? 'reset' : '');
```

Comment:
:art: I think we call in the most other cases the observable with `undefined` (or rather no parameter) if we don't want to reset it. Maybe it makes sense to also call it here with `undefined` instead of `''` so we can type it as `'reset' | undefined`.